src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -158,6 +166,26 @@ def _handle_mouse_click_event(
                 success=True,
             )
 
+    def _handle_key_event(self, key: Qt.Key) -> None:
+        if self.reshape_mode is True:
+            if key == Qt.Key_Escape or (
+                key == Qt.Key_Backspace
+                and self.new_segment_rubber_band.numberOfVertices() == 0
+            ):
+                self.new_segment_rubber_band.reset()
+                self.temporary_new_segment_rubber_band.reset()
+                self._change_to_pick_location_mode()
+                return
+            elif key in (Qt.Key_Backspace, Qt.Key_Delete):
+                if self.new_segment_rubber_band.numberOfVertices() > 1:
+                    self.new_segment_rubber_band.removeLastPoint(0, True)
+                self.temporary_new_segment_rubber_band.reset()
+                self.temporary_new_segment_rubber_band.addPoint(
```

Comment:
could this logic work without moving the cursor after pressing delete, ie. temp segment is from point n -> cursor, delete is pressed and point n is removed, afterwards temp segment should be from point (n-1) -> cursor?

Reply:
Yes it could. I did not get it working earlier, but there might have been numerous reasons for that.

I'll try to fix this.